packages/goto/scripts/postinstall.js

```diff
@@ -1,54 +1,116 @@
   'https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/privacy.txt',
   'https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/resource-abuse.txt',
   'https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/unbreak.txt',
-  'https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/annoyances.txt',
-
-  // Easylist – https://easylist.to/
-  'https://easylist.to/easylist/easylist.txt',
-  'https://easylist.to/easylist/easyprivacy.txt',
-  'https://easylist.to/easylist/fanboy-annoyance.txt',
-  'https://easylist.to/easylist/fanboy-social.txt',
 
+  // Fanboy – https://www.fanboy.co.nz
+  // Easylist, Easyprivacy, Enhanced Trackers
+  'https://www.fanboy.co.nz/r/fanboy-complete.txt',
+  // Fanboy Annoyances List + Fanboy-Social List
+  'https://easylist-downloads.adblockplus.org/fanboy-annoyance.txt',
   // Other
-  'http://pgl.yoyo.org/as/serverlist.php?hostformat=adblockplus;showintro=0&mimetype=plaintext',
-  'https://fanboy.co.nz/fanboy-cookiemonster.txt',
-  'https://raw.githubusercontent.com/liamja/Prebake/master/obtrusive.txt'
+  // Cookies Lists
+  'https://www.fanboy.co.nz/fanboy-cookiemonster.txt',
+  // crypto miners
+  'https://raw.githubusercontent.com/hoshsadiq/adblock-nocoin-list/master/nocoin.txt'
+]
+
+const HOSTS = [
+  // Steven Black hosts – https://github.com/StevenBlack/hosts
+  'https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts'
 ]
 
-const rulesFromURL = async url => {
-  const { body } = await got(url)
-  return body
+const OUTPUT_FILENAME = 'src/engine.bin'
+
+const fetch = async url => (await got(url)).body
+
+const fetchLists = async (urls, fn) => {
+  const filterLists = await Promise.all(urls.map(fn))
+  const set = filterLists.reduce((acc, set) => new Set([...acc, ...set]), new Set())
+  debug(fn.type, { urls: urls.length, count: set.size })
+  return Array.from(set)
 }
 
-const main = async urls => {
-  const rules = Array.from(
-    getLinesWithFilters((await Promise.all(urls.map(rulesFromURL))).join(EOL))
-  )
-  const engine = PuppeteerBlocker.parse(rules.join(EOL))
-  engine.updateResources(await rulesFromURL(RESOURCES), '' + Date.now())
-  await writeFile('src/engine.bin', engine.serialize())
+const fetchFilterList = async url => {
+  const entries = await fetch(url)
+  const filters = getLinesWithFilters(entries)
+  debug('filters', { url, count: filters.size })
+  return filters
 }
 
-main(FILTERS)
-  .catch(err => {
-    console.error(err)
-    process.exit(1)
+fetchFilterList.type = 'filters'
+
+const fetchHostList = async url =>
+  new Promise((resolve, reject) => {
+    const hosts = []
+    got
+      .stream(url)
+      .pipe(split())
+      .on('error', reject)
+      .on('finish', () => {
+        debug('hosts', { url, count: hosts.length })
+        return resolve(getLinesWithFilters(hosts.join(EOL)))
+      })
+      .on('data', data => {
+        let line = data.toString()
```

Comment:
Why is it needed to call `toString(...)` on `data`. Isn't it already a `string`?

Reply:
It's a `Buffer`; It needs to do `toString` in order to serialize the value.